q2_metadata/_normalize.py

```diff
@@ -27,27 +28,37 @@ def normalize(metadata: q2.Metadata, rules_dir: q2.plugin.Str) -> q2.Metadata:
 
     Returns
     -------
-    metadata_curated : q2.Metadata
+    metadata_curated : pd.DataFrame
         Curated metadata table.
     """
 
     # TEMPORARY FUNCTION TO PASS THE DEFAULT FOLDER CONTAINING OUR 8 RULES
     # (A REAL USER SHOULD PASS ANOTHER FOLDER LOCATION TO '--p-rules-dir')
     variables_rules_dir = get_variables_rules_dir(rules_dir, RULES)
 
+    # initialize the collection objects as well as for warning and errors
+    rules = RulesCollection()
+    warnings = WarningsCollection()
+    errors = ErrorsCollection()
+
     # Collect rules from yaml files folder by instantiating a class
-    rules = RulesCollection(variables_rules_dir)
+    variables_rules_files = rules.check_variables_rules_dir(variables_rules_dir)
+
+    # parse all the variables' yaml rules files
+    rules.parse_variables_rules(list(variables_rules_files))
 
     # Get metadata as pandas data frame
     md = metadata.to_dataframe()
 
     # get metadata variables that have rules
-    focus = get_intersection(rules.get_variables_names(), md.columns.tolist())
+    focus = get_intersection(rules.variables_rules.keys(), md.columns.tolist())
+
+    # checks correct rules format and put rules in data structure
+    rules.check_variables_rules(focus)
```

Comment:
The parsing is just a basic yaml.load(). In a previous review, Rebecca, @ElDeveloper and I decided to remove a series of exceptions that I coded as a proposition to catch yaml parsing error. Indeed, if there is an error at the basic parsing step (see call to method `rules.parse_variables_rules(variables_rules_files)` in the main), that's not really the resort of the normalization plugin...

In fact, I have been defining the methods and then calling them from the main because I thought doing the calls from within `RulesCollection` would alleviate what you called `cyclomatic complexity` (I tried to find what that meant and I understood that it could be related to calling functions inside function inside functions etc...). Since the original code was like you now suggest (i.e. not only the above-comment `check_variable_rules_dir`, but also this `check_variables_rules` method was called "automatically" in the class `__init__()`), and that after initial critics and changes that with Rebecca we worked to rather make a step by step methods calls from the main `_normalize()`, I am now confused.

To me, this is still the responsibility of `RulesCollection`, since these are methods of this class.